roles/kubernetes/master/templates/kubeadm-config.v1alpha1.yaml.j2

```diff
@@ -20,8 +20,11 @@ networking:
   serviceSubnet: {{ kube_service_addresses }}
   podSubnet: {{ kube_pods_subnet }}
 kubernetesVersion: {{ kube_version }}
-{% if cloud_provider is defined and cloud_provider not in ["gce", "oci"] %}
-cloudProvider: {{ cloud_provider }}
+{% if cloud_provider is defined and cloud_provider in ["openstack", "azure", "vsphere"] %}
+  cloudProvider: {{cloud_provider}}
+  cloudConfig: {{ kube_config_dir }}/cloud_config
+{% elif cloud_provider is defined and cloud_provider in ["aws", "external"] %}
```

Comment:
shoud be move cloud provider aws to 23 line.

Reply:
and all configure need  move to apiserverExtraArgs.

Reply:
Are you sure about that in v1alpha1 ?